GenderPayGap.WebJob/Functions/Functions.Purge.cs

```diff
@@ -27,48 +28,71 @@ public async Task PurgeUsers([TimerTrigger("40 3 * * *" /* 03:40 once per day */
+                    uo => uo.PINConfirmedDate != null || (uo.PINSentDate != null && uo.PINSentDate < pinExpiryDate)))
                 {
-                    //Ignore if they have verified PIN
-                    if (user.UserOrganisations.Any(
-                        uo => uo.PINConfirmedDate != null || uo.PINSentDate != null && uo.PINSentDate < pinExpireyDate))
-                    {
-                        continue;
-                    }
-
-                    var logItem = new ManualChangeLogModel(
-                        nameof(PurgeUsers),
-                        ManualActions.Delete,
-                        AppDomain.CurrentDomain.FriendlyName,
-                        nameof(user.UserId),
-                        user.UserId.ToString(),
-                        null,
-                        JsonConvert.SerializeObject(new {user.UserId, user.EmailAddress, user.JobTitle, user.Fullname}),
-                        null);
-                    _DataRepository.Delete(user);
-                    await _DataRepository.SaveChangesAsync();
-                    await Global.ManualChangeLog.WriteAsync(logItem);
+                    return;
                 }
 
-                LogFunctionEnd(runId, nameof(PurgeUsers), startTime);
+                var logItem = new ManualChangeLogModel(
+                    nameof(PurgeUsers),
+                    ManualActions.Delete,
+                    AppDomain.CurrentDomain.FriendlyName,
+                    nameof(user.UserId),
+                    user.UserId.ToString(),
+                    null,
+                    JsonConvert.SerializeObject(new {user.UserId, user.EmailAddress, user.JobTitle, user.Fullname}),
+                    null);
+            
+                DeleteUserAndAuditLogs(user);
+                await Global.ManualChangeLog.WriteAsync(logItem);
             }
             catch (Exception ex)
             {
                 LogFunctionError(runId, nameof(PurgeUsers), startTime, ex );
+            }
 
-                //Send Email to GEO reporting errors
-                await _Messenger.SendGeoMessageAsync("GPG - WEBJOBS ERROR", 
-                    $"Failed webjob ({nameof(PurgeUsers)}):{ex.Message}:{ex.GetDetailsText()}");
-                //Rethrow the error
-                throw;
+        }
+
+        private static void DeleteUserAndAuditLogs(User user)
+        {
+            var dataRepository = Program.ContainerIOC.Resolve<IDataRepository>();
+            
+            List<AuditLog> auditLogs = dataRepository
+                .GetAll<AuditLog>()
+                .Where(log => Equals(log.ImpersonatedUser, user))
+                .Where(log => log.Action == AuditedAction.AdminResendVerificationEmail).ToList();
+            
+            foreach (AuditLog log in auditLogs)
+            {
+                dataRepository.Delete(log);
             }
+            
+            dataRepository.Delete(user);
+            dataRepository.SaveChangesAsync().Wait();

```

Comment:
I haven't been able to test this locally, and I'm not 100% sure this will work. I wonder if I need to try 

`foreach (AuditLog log in auditLogs)
            {
                dataRepository.Delete(log);
            }
            dataRepository.SaveChangesAsync().Wait();
            dataRepository.Delete(user);
            dataRepository.SaveChangesAsync().Wait();`

Reply:
Good spot. I'm not sure - it's worth trying

Reply:
It works the way it is